Editor/Scenes/SongEditor/SongEditor.gd

```diff
@@ -34,15 +45,46 @@ func add_track(instrument: Button):
 	var inst := GoDAW.get_instrument(instrument.text)
 	$Sequencer.INSTRUMENTS[instrument.text] = inst
 
-func sequence():
+func check_error(script_location: String) -> String:
+	# Error check
+	var err = []
+	var _n = OS.execute(BIN, ["-s", script_location, "--check-only", "--no-window"], true, err, true)
+	var error = (err[0] as String).substr(err[0].find("SCRIPT"))\
+		.replace(script_location, "SongScript")\
+		.replace(" (", "")\
+		.replace("SCRIPT ERROR: ", "")\
+		.replace("GDScript::reload", "")
+	var prev = 0
+	for i in error.count("SongScript:"):
+		prev = error.find(")", error.find("SongScript"))
+		error[prev] = ""
+	return error
+
+# Return true if everything goes alright
+func sequence() -> bool:
 	if !gui:
 		var file =  File.new()
-		var dir = Directory.new()
-		file.open("res://song.gd", File.WRITE)
-		file.store_string(song_script_editor.text)
+		file.open("user://song.gd", File.WRITE_READ)
+		if in_file != song_script_editor.text:
+			file.store_string(song_script_editor.text)
+#			yield(get_tree(), "idle_frame")
+			in_file = file.get_as_text() # Idk why but the file isn't storing without this Wierd
```

Comment:
I'd assume the file has to be closed after you wrote to it :)

Reply:
hmm

Reply:
Browsing through the docs said i had to flush